lib/workers/repository/init/merge.ts

```diff
@@ -237,6 +239,7 @@ export async function mergeRenovateConfig(
     logger.trace({ config: resolvedConfig }, 'resolved config after migrating');
     resolvedConfig = migrationResult.migratedConfig;
   }
+  setNpmTokenInNpmrc(resolvedConfig);
```

Comment:
why do we need to do it twice instead once at the end?

Reply:
npmrc is being set at these 2 instances. So its necessary to do so else we will save `npmrc` string in the `.npmrc` file which does not contain the token.